sympy/integrals/risch.py

```diff
@@ -433,6 +443,41 @@ def _auto_attrs(self):
         self.d = self.D[self.level]
         self.case = self.cases[self.level]
 
+    def _tan_part(self, tans):
+
+        new_extension = False
+        restart = False
+        tanargs = [i.args[0] for i in tans]
+        ip = integer_powers(tanargs)
+
+        for arg, others in ip:
+            others.sort(key=lambda i: i[1])
+
+            arga, argd = frac_in(arg, self.t)
```

Comment:
Was stepping through this in a debugger. These are not rewritten in terms of the extensions so far (like if you have `tan(log(x))` then `arg` will be `log(x)` instead of `t0`). Is that correct?